cmd/istio-cni/main.go

```diff
@@ -169,9 +171,22 @@ func cmdAdd(args *skel.CmdArgs) error {
 				return err
 			}
 			log.Debug("Created Kubernetes client", zap.Reflect("client", client))
-			containers, initContainersMap, _, annotations, ports, k8sErr := getKubePodInfo(client, string(k8sArgs.K8S_POD_NAME), string(k8sArgs.K8S_POD_NAMESPACE))
+			var containers []string
+			var initContainersMap map[string]struct{}
+			var annotations map[string]string
+			var ports []string
+			var k8sErr error
+			for retry := 0; retry < podRetrievalMaxRetries; retry++ {
+				containers, initContainersMap, _, annotations, ports, k8sErr = getKubePodInfo(client, string(k8sArgs.K8S_POD_NAME), string(k8sArgs.K8S_POD_NAMESPACE))
+				if k8sErr == nil {
+					break
+				}
+				log.Warnf("Waiting for pod data %v", k8sErr)
+				time.Sleep(10 * time.Second)
```

Comment:
This is too long. IIRC, CNI has a timeout, so we should try to return ASAP.
Set this to something like 1 or 2 seconds?

Reply:
Done. Changed to 1 second.